internal/hcs/system.go

```diff
@@ -326,13 +336,113 @@ func (computeSystem *System) Properties(ctx context.Context, types ...schema1.Pr
 	return properties, nil
 }
 
+// statisticsInProc emulates what HCS does to grab statistics for a given container with a small
+// change to make grabbing the private working set total much more efficient.
+func (computeSystem *System) statisticsInProc(ctx context.Context) (*hcsschema.Properties, error) {
+	// See if we'd even be able to open the silo. We'll need to be local
+	// system/system so check first.
+	usr, err := windows.GetCurrentProcessToken().GetTokenUser()
+	if err != nil {
+		return nil, err
+	}
+	if !usr.User.Sid.IsWellKnown(windows.WinLocalSystemSid) {
+		return nil, errors.New("process does not have the right permissions to open the silo")
+	}
+
+	jobOptions := &jobobject.Options{
+		UseNTVariant: true,
+		Name:         siloNameFmt(computeSystem.id),
+	}
+	job, err := jobobject.Open(ctx, jobOptions)
+	if err != nil {
+		return nil, err
+	}
+	defer job.Close()
+
+	memInfo, err := job.QueryMemoryStats()
+	if err != nil {
+		return nil, err
+	}
+
+	processorInfo, err := job.QueryProcessorStats()
+	if err != nil {
+		return nil, err
+	}
+
+	storageInfo, err := job.QueryStorageStats()
+	if err != nil {
+		return nil, err
+	}
+
+	privateWorkingSet, err := job.PrivateWorkingSet()
+	if err != nil {
+		return nil, err
+	}
+
+	return &hcsschema.Properties{
+		Statistics: &hcsschema.Statistics{
+			Timestamp:          time.Now(),
+			ContainerStartTime: computeSystem.startTime,
+			Uptime100ns:        uint64(time.Since(computeSystem.startTime)) / 100,
```

Comment:
Much prefer that we explicitly include a units conversion here, rather than assuming that `time.Duration` is nanoseconds.

Reply:
Sure, let me fix this for the jobcontainer stats as well then

Reply:
Done, resolving